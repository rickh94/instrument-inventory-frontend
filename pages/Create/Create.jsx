import React, { useState, useContext } from 'react'
import PropTypes from 'prop-types'

import { API } from 'aws-amplify'
import { LoadingHeader, RootPaper, SchemaForm } from '../../components'
import temporaryError from '../../libs/temporaryError'
import { HelpersContext, SchemaContext } from '../../contexts'

const Create = ({ history }) => {
  const { showAlert } = useContext(HelpersContext)
  const schema = useContext(SchemaContext)
  const [isLoading, setLoading] = useState(false)
  const [error, setError] = useState(null)

  const handleSubmit = async body => {
    try {
      setLoading(true)
      const response = await API.post('instrument-inventory', 'instruments', { body })
      setLoading(false)
      history.push(`/instrument/${response.id}`)
      showAlert(`${response.item.type} ${response.item.number} created`)
    } catch (err) {
      temporaryError({ setError, message: err })
      setLoading(false)
    }
  }

  return (
    <React.Fragment>
      <RootPaper>
        <LoadingHeader isLoading={isLoading} title="Create a New Instrument" />
        {schema && (
          <SchemaForm
            schema={schema.components.schemas.Instrument}
            onSubmit={handleSubmit}
            error={error}
          />
        )}
      </RootPaper>
    </React.Fragment>
  )
}

Create.propTypes = {
  history: PropTypes.object.isRequired,
}

export default Create
