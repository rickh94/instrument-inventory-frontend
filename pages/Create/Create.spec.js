import React from 'react'
import Create from './Create'
import { render } from '@testing-library/react'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

const schema = {
  components: {
    schemas: {
      Instrument: {
        type: 'object',
        properties: {
          number: {
            title: 'Instrument Number',
            description: 'The inventory number of the instrument',
            type: 'string',
          },
          size: {
            title: 'Instrument Size',
            description: 'The fractional or inch size of the instrument',
            enum: [
              '1/16',
              '1/10',
              '1/8',
              '1/4',
              '1/2',
              '3/4',
              '7/8',
              '4/4',
              '9"',
              '10"',
              '11"',
              '12"',
              '13"',
              '14"',
              '15"',
              '15.5"',
              '16"',
              '16.5"',
              '17"',
            ],
            type: 'string',
          },
          type: {
            title: 'Instrument Type',
            description: 'What kind of instrument',
            enum: ['Violin', 'Viola', 'Violin strung as viola', 'Cello', 'Bass'],
            type: 'string',
          },
          location: {
            title: 'Instrument Location',
            enum: [
              'Grant Elementary School',
              'Hedgepath Middle School',
              'Wilson Elementary School',
              'Trenton High School',
              'Columbus Elementary School',
              'Office',
              'Storage',
              'Trade',
              'Maintenance',
              'Transit',
              'Unknown',
            ],
            type: 'string',
          },
          assignedTo: {
            title: 'Assigned To',
            description: 'Who it is signed out to',
            type: 'string',
          },
          maintenanceNotes: {
            title: 'Maintenance Notes',
            maxLength: 200,
            type: 'string',
          },
          conditionNotes: {
            title: 'Condition Notes',
            maxLength: 200,
            type: 'string',
          },
          condition: {
            title: 'Condition',
            exclusiveMaximum: 6,
            gte: 0,
            type: 'integer',
          },
          quality: {
            title: 'Quality',
            exclusiveMaximum: 6,
            gte: 0,
            type: 'integer',
          },
          gifted: {
            title: 'Gifted To Student',
            description:
              'Whether it has been given permanently to the student it is assigned to.',
            default: false,
            type: 'boolean',
          },
        },
      },
    },
  },
}

describe('<Create />', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <MuiThemeProvider theme={createMuiTheme()}>
        <Create schema={schema} history={{ push: jest.fn() }} showAlert={jest.fn()} />
      </MuiThemeProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
