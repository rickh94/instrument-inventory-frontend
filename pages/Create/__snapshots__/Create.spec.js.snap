// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create /> matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded WithStyles(ForwardRef(Paper))-root-1"
  >
    <div
      class="makeStyles-heading-29"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
      >
        Create a New Instrument
      </h5>
    </div>
    <form
      novalidate=""
    >
      <div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            for="scanner-field"
          >
            Instrument Number
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
          >
            <input
              aria-describedby="scanner-error"
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
              id="scanner-field"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="scan-button"
                tabindex="0"
                title="Scan Barcode"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-barcode fa-w-16 "
                    data-icon="barcode"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 448V64h18v384H0zm26.857-.273V64H36v383.727h-9.143zm27.143 0V64h8.857v383.727H54zm44.857 0V64h8.857v383.727h-8.857zm36 0V64h17.714v383.727h-17.714zm44.857 0V64h8.857v383.727h-8.857zm18 0V64h8.857v383.727h-8.857zm18 0V64h8.857v383.727h-8.857zm35.715 0V64h18v383.727h-18zm44.857 0V64h18v383.727h-18zm35.999 0V64h18.001v383.727h-18.001zm36.001 0V64h18.001v383.727h-18.001zm26.857 0V64h18v383.727h-18zm45.143 0V64h26.857v383.727h-26.857zm35.714 0V64h9.143v383.727H476zm18 .273V64h18v384h-18z"
                      fill="currentColor"
                    />
                  </svg>
                  <span
                    class="MuiTypography-root MuiTypography-srOnly"
                    data-testid="sr-text"
                  >
                    Scan Barcode
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
            data-shrink="false"
          >
            Instrument Size
          </label>
          <div
            appearance="checkbox"
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <div
              class="MuiSelect-root"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                class="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                id="select-size"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                id="uniforms-0000-0003"
                name="size"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
            data-shrink="false"
          >
            Instrument Type
          </label>
          <div
            appearance="checkbox"
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <div
              class="MuiSelect-root"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                class="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                id="select-type"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                id="uniforms-0000-0005"
                name="type"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
            data-shrink="false"
          >
            Instrument Location
          </label>
          <div
            appearance="checkbox"
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <div
              class="MuiSelect-root"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                class="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                id="select-location"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                id="uniforms-0000-0007"
                name="location"
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
            data-shrink="false"
            for="uniforms-0000-0009"
          >
            Assigned To
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="uniforms-0000-0009"
              name="assignedTo"
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
            data-shrink="false"
            for="uniforms-0000-000b"
          >
            Maintenance Notes
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="uniforms-0000-000b"
              name="maintenanceNotes"
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense"
            data-shrink="false"
            for="uniforms-0000-000d"
          >
            Condition Notes
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
              id="uniforms-0000-000d"
              name="conditionNotes"
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            for="rating-field"
          >
            Condition
          </label>
          <div
            class="Rating-root-159"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            for="rating-field"
          >
            Quality
          </label>
          <div
            class="Rating-root-159"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiIconButton-root Rating-iconButton-160"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root Rating-icon-161"
                  focusable="false"
                  nativeColor="#e0e0e0"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <fieldset
          class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth"
        >
          <div
            class="MuiFormGroup-root"
          >
            <label
              class="MuiFormControlLabel-root"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary PrivateSwitchBase-root-178 MuiCheckbox-root MuiCheckbox-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <input
                    class="PrivateSwitchBase-input-181"
                    data-indeterminate="false"
                    id="uniforms-0000-000j"
                    name="gifted"
                    type="checkbox"
                    value="gifted"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
              >
                Gifted To Student
              </span>
            </label>
          </div>
        </fieldset>
      </div>
      <div
        class="MuiFormGroup-root MuiFormGroup-row"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-button-61 makeStyles-lastButton-62 MuiButton-textPrimary MuiButton-text"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
